src/helpers.cpp

```diff
@@ -539,8 +545,8 @@ ada_really_inline void parse_prepared_path(std::string_view input,
               ? path_buffer_tmp
               : path_view;
       if (unicode::is_double_dot_path_segment(path_buffer)) {
-        helpers::shorten_path(path, type);
-        if (location == std::string_view::npos) {
+        if ((helpers::shorten_path(path, type) || special) &&
```

Comment:
would changing the order and moving special to the first expression improve performance?

Reply:
I think that moving the order would change the logic and break the code because the compiler would be allowed not to try to shorten the path.